core/src/commonMain/kotlin/com/xebia/functional/xef/auto/AIRuntime.kt

```diff
@@ -0,0 +1,32 @@
+package com.xebia.functional.xef.auto
+
+import com.xebia.functional.xef.embeddings.OpenAIEmbeddings
+import com.xebia.functional.xef.env.OpenAIConfig
+import com.xebia.functional.xef.llm.openai.KtorOpenAIClient
+import com.xebia.functional.xef.llm.openai.LLMModel
+import com.xebia.functional.xef.vectorstores.LocalVectorStore
+import kotlin.jvm.JvmStatic
+import kotlin.time.ExperimentalTime
+
+data class AIRuntime<A>(val runtime: suspend (block: AI<A>) -> A) {
+  companion object {
+    @OptIn(ExperimentalTime::class)
+    @JvmStatic
+    fun <A> openAI(): AIRuntime<A> = AIRuntime { block ->
+      val openAIConfig = OpenAIConfig()
+      KtorOpenAIClient(openAIConfig).use { openAiClient ->
```

Comment:
We need to redesign this part to prevent creating an instance of the HTTP client for every `ai` block

Reply:
We will get rid of our custom Ktor implementation in favor of https://github.com/aallam/openai-kotlin, which just added function support. All our client-related code for open ai will then be gone. that lib keeps up with the entire API, including the missing methods for images and others.